Android doesn't have getdtablesize. Use sysconf(_SC_OPEN_MAX) instead.

diff -ur qemu-2.9.0-orig/slirp/misc.c qemu-2.9.0/slirp/misc.c
--- qemu-2.9.0-orig/slirp/misc.c	2017-07-31 04:13:25.711190162 +0200
+++ qemu-2.9.0/slirp/misc.c	2017-07-31 05:24:42.082381735 +0200
@@ -144,7 +144,7 @@
 		dup2(s, 0);
 		dup2(s, 1);
 		dup2(s, 2);
-		for (s = getdtablesize() - 1; s >= 3; s--)
+		for (s = sysconf(_SC_OPEN_MAX) - 1; s >= 3; s--)
 		   close(s);
 
 		i = 0;
